canary/src/main/java/io/littlehorse/canary/kafka/MetricsEmitter.java

```diff
@@ -0,0 +1,52 @@
+package io.littlehorse.canary.kafka;
+
+import io.littlehorse.canary.CanaryException;
+import io.littlehorse.canary.proto.Metric;
+import java.io.Closeable;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.utils.Bytes;
+
+@Slf4j
+public class MetricsEmitter implements Closeable {
+
+    private final Producer<String, Bytes> producer;
+    private final String topicName;
+
+    public MetricsEmitter(final String topicName, final Map<String, Object> kafkaProducerConfigMap) {
+        this.producer = new KafkaProducer<>(kafkaProducerConfigMap);
+        this.topicName = topicName;
+    }
+
+    public Future<RecordMetadata> future(final String key, final Metric metric) {
+        final ProducerRecord<String, Bytes> record =
+                new ProducerRecord<>(topicName, key, Bytes.wrap(metric.toByteArray()));
+
+        return producer.send(record, (metadata, exception) -> {
+            if (exception == null) {
+                log.debug("Emitting message {} {}", metric.getMetricCase(), key);
+            } else {
+                log.error("Emitting message {} {}", metric.getMetricCase(), key, exception);
+            }
+        });
+    }
+
+    public RecordMetadata emit(final String key, final Metric metric) {
```

Comment:
Could you put in the Javadoc on this method that it is blocking? Additionally, we may want to be careful about _where_ we call this method. For example, probably we shoudn't call it from within the Aggregator Streams Topology.

However, if we call it from the Metronome Task Worker, that is fine (in fact, we _need_ to make it blocking for that use-case).